### PR TITLE
feat(components/phone-field): add phone field test harness

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**
[AB#2195518](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2195518)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2849/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2849/storybooks/ag-grid-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2849/storybooks/phone-field-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2849/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2849/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2849/playground/)

File: libs/components/phone-field/testing/src/phonefield/phone-field-harness.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import { SkyPhoneFieldInputHarness } from './phone-field-input-harness';
+
+/**
+ * Harness for interacting with a phone field component in tests.
+ */
+export class SkyPhoneFieldHarness extends SkyComponentHarness {
+  /**
+   * Finds a standard phone field component, or a phone field component that is wrapped by an input box component.
+   * For input box implementations, we need to use the `.sky-input-box` selector since the `sky-country-field`
+   * element is removed from the DOM.
```
Comment (Blackbaud-SandhyaRajasabeson) [inline libs/components/phone-field/testing/src/phonefield/phone-field-harness.ts:15]: can we use the new text that is used in lookup and datepicker?

File: libs/components/phone-field/testing/src/phonefield/phone-field-harness.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  ): HarnessPredicate<SkyPhoneFieldHarness> {
+    return SkyPhoneFieldHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /**
+   * Blurs the phone field input.
+   */
+  public async blur(): Promise<void> {
+    return (await this.#getInput()).blur();
+  }
```
Comment (Blackbaud-SandhyaRajasabeson) [inline libs/components/phone-field/testing/src/phonefield/phone-field-harness.ts]: consumers own the input so i dont think we need to add functions that manipulate the input

File: libs/components/phone-field/testing/src/phonefield/phone-field-harness.ts (hunk 1 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/phone-field/testing/src/phonefield/phone-field-harness.ts:15]: added

File: libs/components/phone-field/testing/src/phonefield/phone-field-harness.ts (hunk 2 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/phone-field/testing/src/phonefield/phone-field-harness.ts]: updated to use new child input harness pattern

File: apps/code-examples/src/app/code-examples/phone-field/phone-field/basic/demo.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import { SkyPhoneFieldHarness } from '@skyux/phone-field/testing';
+
+import { DemoComponent } from './demo.component';
+
+const COUNTRY_AU: SkyPhoneFieldCountry = {
+  name: 'Australia',
+  iso2: 'au',
+  dialCode: '+61',
+};
+// const COUNTRY_US: SkyPhoneFieldCountry = {
```
Comment (Blackbaud-TrevorBurch) [inline apps/code-examples/src/app/code-examples/phone-field/phone-field/basic/demo.component.spec.ts]: Is this commented code needed?

File: apps/code-examples/src/app/code-examples/phone-field/phone-field/basic/demo.component.ts [hunk 4]
```diff
@@ -21,15 +22,17 @@ import { SkyPhoneFieldModule } from '@skyux/phone-field';
   ],
 })
 export class DemoComponent {
-  protected phoneForm: FormGroup;
+  public phoneForm: FormGroup;
```
Comment (Blackbaud-TrevorBurch) [inline apps/code-examples/src/app/code-examples/phone-field/phone-field/basic/demo.component.ts:25]: Why are these changing to be `public`?

Comment (Blackbaud-CoreyArcher) [reply apps/code-examples/src/app/code-examples/phone-field/phone-field/basic/demo.component.ts:25]: We have been making form controls or groups public to be accessible in tests.

File: apps/code-examples/src/app/code-examples/phone-field/phone-field/basic/demo.component.spec.ts (hunk 3 above)
Comment (Blackbaud-CoreyArcher) [reply apps/code-examples/src/app/code-examples/phone-field/phone-field/basic/demo.component.spec.ts]: no. removed

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Pull request merged**